django/core/checks/security/base.py

```diff
@@ -69,8 +69,9 @@
 
 W009 = Warning(
     "Your SECRET_KEY has less than %(min_length)s characters or less than "
-    "%(min_unique_chars)s unique characters. Please generate a long and random "
-    "SECRET_KEY, otherwise many of Django's security-critical features will be "
+    "%(min_unique_chars)s unique characters, or it's a Django-generated secret key starting with django-insecure."
+    "Please generate a sufficiently long and random SECRET_KEY, "
+    "otherwise many of Django's security-critical features will be "
     "vulnerable to attack." % {
```

Comment:
Things seem to be hung up on "Django-generated". I propose the following wording instead which seems clearer to me. I also note that "sufficiently" sneaked in there, but I'm not sure it really adds anything. Also note that the line wrapping should be maintained.

```suggestion
    "Your SECRET_KEY has less than %(min_length)s characters, less than "
    "%(min_unique_chars)s unique characters, or it is prefixed with "
    "'django-insecure-' indicating that it was generated automatically by "
    "Django. Please generate a long and random SECRET_KEY, otherwise many of "
    "Django's security-critical features will be vulnerable to attack." % {
```

(Note this suggestion may not apply cleanly in GitHub as I needed to include unchanged lines around the removed and added lines.)